providers/gcp/gcp.go

```diff
@@ -213,12 +218,28 @@ func (p *Provider) addProductPath(product string) {
 }
 
 func (p *Provider) addProjectParam() {
-	params := url.Values{}
-	params.Add("project", p.SDKConfig.Core.Project)
-	p.URL.RawQuery = params.Encode()
+	q := p.URL.Query()
+	q.Add("project", p.SDKConfig.Core.Project)
+	p.URL.RawQuery = q.Encode()
+	return
+}
+
+func (p *Provider) addGKEPageStateParam(namespace string) {
+	q := p.URL.Query()
+	q.Add("pageState", constructPageStateParam(namespace))
+	p.URL.RawQuery = q.Encode()
 	return
 }
 
+func constructPageStateParam(namespace string) string {
+	nss := strings.Split(namespace, ",")
+	quotedNSs := make([]string, 0, len(nss))
+	for _, s := range nss {
+		quotedNSs = append(quotedNSs, strconv.Quote(s))
+	}
+	return fmt.Sprintf("(\"savedViews\":(\"n\":[%s]))", strings.Join(quotedNSs, ","))
+}
+
```

Comment:
I think `nss` might be confusing, may I ask you to use the `namespaces`, please 🙏 

```suggestion
func constructPageStateParam(rawNamespaces string) string {
	namespaces := strings.Split(rawNamespaces, ",")
	quotedNamespaces := make([]string, 0, len(namespaces))
	for _, s := range namespaces {
		quotedNamespaces = append(quotedNSs, strconv.Quote(s))
	}
	return fmt.Sprintf("(\"savedViews\":(\"n\":[%s]))", strings.Join(quotedNamespaces, ","))
}

```